common/src/main/java/com/scottlogic/deg/common/profile/Types.java

```diff
@@ -2,19 +2,20 @@
 
 import java.time.OffsetDateTime;
 import java.util.function.Function;
+import java.util.function.Predicate;
 
 public enum Types {
     NUMERIC(o -> o instanceof Number),
     STRING(o -> o instanceof String),
     DATETIME(o -> o instanceof OffsetDateTime);
 
-    private final Function<Object, Boolean> isInstanceOf;
+    private final Predicate<Object> isInstanceOf;
 
-    Types(final Function<Object, Boolean> isInstanceOf) {
+    Types(final Predicate<Object> isInstanceOf) {
```

Comment:
this should just take in a class, and the isInstanceOf method, call class.isInstance(o)

also, 'number' should be BigDecimal now (unrelated to your changes)